tsdb/series_index.go

```diff
@@ -261,6 +263,11 @@ func (idx *SeriesIndex) Clone() *SeriesIndex {
 		tombstones[id] = struct{}{}
 	}
 
+	idOffsetMap := make(map[uint64]int64)
+	for k, v := range idx.idOffsetMap {
+		idOffsetMap[k] = v
+	}
```

Comment:
Since we now check for existence in the map & tombstones, we need to copy over the id/offset map during `Clone()` (which is used to snapshot the current index state).